lib/matplotlib/tests/test_figure.py

```diff
@@ -532,6 +532,13 @@ def test_savefig_pixel_ratio(backend):
     assert ratio1 == ratio2
 
 
+def test_savefig_preserve_layout_engine(tmp_path):
```

Comment:
This worked before didn't it?  It was when the rcParam was set that the problem was?  Or was the layout still getting set to None, and then falling back to the rcParam?  

Otherwise if this works, I'm all for it!  I do wish all the context switching were simpler to debug, but...

Reply:
On `main` this test throws an `AttributeError` because the layout engine got set to `None`.  If we set the rcParam then we would get an assertion error on `main`.  I'm happy to test either or both.